Wikipedia/Code/WMFAppViewController.m

```diff
@@ -1672,6 +1672,10 @@ - (BOOL)gestureRecognizer:(UIGestureRecognizer *)gestureRecognizer shouldRecogni
 
 // The method will be called on the delegate only if the application is in the foreground. If the method is not implemented or the handler is not called in a timely manner then the notification will not be presented. The application can choose to have the notification presented as a sound, badge, alert and/or in the notification list. This decision should be based on whether the information in the notification is otherwise visible to the user.
 - (void)userNotificationCenter:(UNUserNotificationCenter *)center willPresentNotification:(UNNotification *)notification withCompletionHandler:(void (^)(UNNotificationPresentationOptions options))completionHandler {
+    if([notification.request.content.threadIdentifier isEqualToString:EchoModelVersion.current]) {
+        [NSNotificationCenter.defaultCenter postNotificationName:NSNotification.pushNotificationBannerDidDisplayInForeground object:nil userInfo:notification.request.content.userInfo];
+    }
+
     completionHandler(UNNotificationPresentationOptionAlert);
```

Comment:
@tonisevener Interesting idea - an Echo safety net. I wonder though if it might be best as-is. True that calling the completion with that `none` option will suppress all non-Echo push notifications, but unfortunately it'll also suppress even _locally_ scheduled notifications. It also adds a bit of complexity (from an engineer's perspective) if down the line we embrace local notifications more, having to puzzle with, then hunt down or remember, why it is my locally implemented notifications aren't firing. 

What do you think about leaving it as-is, and only explicitly muting all other notification alerts if we suddenly notice an issue we can't track down?

Reply:
Aah good call, I wasn’t thinking about local notifications. Yeah I’m good with leaving it as-is!